src/App.tsx

```diff
@@ -1,95 +1,144 @@
 import React from 'react';
 import 'bulma/css/bulma.css';
 import './App.scss';
-
-export const goodsFromServer = [
-  'Dumplings',
-  'Carrot',
-  'Eggs',
-  'Ice cream',
-  'Apple',
-  'Bread',
-  'Fish',
-  'Honey',
-  'Jam',
-  'Garlic',
-];
+import classNames from 'classnames';
+import { goodsFromServer } from './goodsFromServer';
+import { GoodList } from './components/GoodList';
 
 enum SortType {
   NONE,
   ALPHABET,
   LENGTH,
 }
 
+interface Good {
+  id: number;
+  name: string;
+}
+
+type State = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
 type ReorderOptions = {
   sortType: SortType,
   isReversed: boolean,
 };
 
-// Use this function in the render method to prepare goods
 export function getReorderedGoods(
-  goods: string[],
+  goods: Good[],
   { sortType, isReversed }: ReorderOptions,
 ) {
-  // To avoid the original array mutation
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  switch (sortType) {
+    case SortType.ALPHABET:
+      visibleGoods.sort((good1, good2) => good1.name.localeCompare(good2.name));
+      break;
+
+    case SortType.LENGTH:
+      visibleGoods.sort((good1, good2) => (
+        good2.name.length - good1.name.length));
```

Comment:
```suggestion
        good1.name.length - good2.name.length));
```
should be vice versa

Reply:
all chenged